Extensions/TokenAcquisition.cs

```diff
@@ -39,14 +39,17 @@ public class TokenAcquisition : ITokenAcquisition
     {
         private AzureADOptions _azureAdOptions;
 
+        private ITokenCacheProvider _tokenCacheProvider;
+
         /// <summary>
         /// Constructor of the TokenAcquisition service. This requires the Azure AD Options to 
         /// configure the confidential client application
         /// </summary>
         /// <param name="options">Options to configure the application</param>
-        public TokenAcquisition(IOptionsMonitor<AzureADOptions> options)
+        public TokenAcquisition(IOptionsMonitor<AzureADOptions> options, ITokenCacheProvider tokenCacheProvider)
```

Comment:
Dependency injection of the ITokenCacheProvider to decouple the token acquisition from the implementation of the token cache serialization